presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -839,6 +851,123 @@ public String toString()
         }
     }
 
+    public static class BytesValues
+            extends AbstractTupleDomainFilter
+    {
+        private final byte[][] values;
+        private final byte[][] hashTable;
+        private final int hashTableSizeMask;
+        private final long[] bloom;
+        private final int bloomSize;
+
+        private BytesValues(byte[][] values, boolean nullAllowed)
+        {
+            super(nullAllowed);
+
+            requireNonNull(values, "values is null");
+            checkArgument(values.length > 1, "values must contain at least 2 entries");
+
+            this.values = values;
+            // Linear hash table size is the next power of two greater
+            // than or equal to number of values * 5.
+            int hashTableSize = Integer.highestOneBit(values.length * 5);
+            hashTableSizeMask = hashTableSize - 1;
+            hashTable = new byte[hashTableSize][];
+            // 8 bits of Bloom filter per slot in hash table. The
```

Comment:
these comment lines are two short; the whole comment fits into a single line; would you make it so?